### PR TITLE
Fix integer overflow in GetL0ThresholdSpeedupCompaction

### PR DESCRIPTION
**Pull request by edouarda**
This will prevent integer overflow to return negative values.

### CODE REVIEW

**Comment by edouarda** [comment]
I think you probably want to squash my diffs because some of my pushes are synchronization pushes with you guys.

File: db/column_family.cc [hunk 1]
```diff
@@ -545,14 +545,34 @@ int GetL0ThresholdSpeedupCompaction(int level0_file_num_compaction_trigger,
...
+
   // 1/4 of the way between L0 compaction trigger threshold and slowdown
   // condition.
+  const int one_fourth_trigger_slowdown =
+      level0_file_num_compaction_trigger +
+      ((level0_slowdown_writes_trigger - level0_file_num_compaction_trigger) /
+       4);
+
+  // overflow protection
+  if (one_fourth_trigger_slowdown < 0) {
```
Comment (yiwu-arbug) [inline db/column_family.cc]: I think one_fourth_trigger_slowdown will never get overflow since level0_slowdown_writes_trigger is its upper bound?

**Comment by edouarda** [comment]
Is the build error related to my changes?

**Comment by yiwu-arbug** [comment]
@edouarda not related, and it should have been fixed.

**Comment by edouarda** [comment]
Ok let me know what you need to merge this.

**Pull request merged**